src/repl/scala/tools/nsc/interpreter/JavapClass.scala

```diff
@@ -134,256 +127,160 @@ class JavapClass(
-        .orFailed (throw new IllegalStateException)
-    }
-    // a result per input
-    private def applyOne(options: Seq[String], filter: Boolean, klass: String, inputs: Seq[Input]): Try[JpResult] =
-      Try {
-        task(options, Seq(klass), inputs).call()
-      } map {
-        case true => JpResult(showable(klass, filter))
-        case _    => JpResult(reporter.reportable())
-      } recoverWith {
-        case e: java.lang.reflect.InvocationTargetException => e.getCause match {
-          case t: IllegalArgumentException => Success(JpResult(t.getMessage)) // bad option
-          case x => Failure(x)
-        }
-      } lastly {
-        reporter.clear()
+    override def hasLocation(location: Location): Boolean =
+      location match {
+        case CLASS_PATH => true
+        case _          => false
       }
-    /** Run the tool. */
-    def apply(options: Seq[String], filter: Boolean)(inputs: Seq[Input]): List[JpResult] = (inputs map {
-      case (klass, Success(_))  => applyOne(options, filter, klass, inputs).get
-      case (_, Failure(e))      => JpResult(e.toString)
-    }).toList orFailed List(noToolError)
   }
-
-  object JavapTool {
-    // >= 1.7
-    val JavapTask    = "com.sun.tools.javap.JavapTask"
-
-    private def hasClass(cl: ScalaClassLoader, cn: String) = cl.tryToInitializeClass[AnyRef](cn).isDefined
-
-    def isAvailable = hasClass(loader, JavapTask)
-
-    /** Select the tool implementation for this platform. */
-    def apply() = {
-      require(isAvailable)
-      new JavapTool
-    }
+  def fileManager(inputs: Seq[Input]) = new JavapFileManager(inputs)()
+
+  // eventually, use the tool interface
+  def task(options: Seq[String], classes: Seq[String], inputs: Seq[Input]): Task = {
+    //ServiceLoader.load(classOf[javax.tools.DisassemblerTool]).
+    //getTask(writer, fileManager, reporter, options.asJava, classes.asJava)
+    val toolopts = options filter (_ != "-filter")
+    loader.create[Task](JavapTask.taskClassName, Console.println(_))(writer, fileManager(inputs), reporter, toolopts.asJava, classes.asJava)
   }
+  // a result per input
+  private def applyOne(options: Seq[String], filter: Boolean, klass: String, inputs: Seq[Input]): Try[JpResult] =
+    Try {
+      task(options, Seq(klass), inputs).call()
+    } map {
+      case true => JpResult(showable(intp, filter, filtered(klass, s"${reporter.reportable()}${written}")))
+      case _    => JpResult(reporter.reportable())
+    } recoverWith {
+      case e: java.lang.reflect.InvocationTargetException => e.getCause match {
+        case t: IllegalArgumentException => Success(JpResult(t.getMessage)) // bad option
+        case x => Failure(x)
+      }
+    } lastly {
+      reporter.clear()
+    }
+  /** Run the tool. */
+  override def apply(options: Seq[String], filter: Boolean)(inputs: Seq[Input]): List[JpResult] = inputs.map {
+    case Input(klass, actual, Success(_)) => applyOne(options, filter, klass, inputs).get
```

Comment:
Here and in the "direct" version, shouldn't it split the "target#pattern", run `javap actual` and filter on pattern?

Reply:
The file manager was looking up by user's target, not the actual path.